### PR TITLE
Elrond - basic version

### PR DESCRIPTION
**Pull request by isra67**

### CODE REVIEW

File: src/model/request/egld/EgldEsdtTransaction.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    @IsOptional()
+    public nonce?: number;
+
+    @IsOptional()
+    public data?: any;
+
+    @IsOptional()
+    @IsNotEmpty()
+    public signatureId?: string;
```
Comment (samuelsramko) [inline src/model/request/egld/EgldEsdtTransaction.ts:28]: this should not be here - its in the PrivateKeyOrSignatureId

File: src/model/request/egld/EsdtToken.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import {IsAlphanumeric, IsIn, IsNotEmpty, IsOptional, IsUppercase, Length, ValidateIf} from 'class-validator';
+
+export class EsdtToken {
+    @IsNotEmpty()
+    @Length(1, 63)
+    @IsIn([
```
Comment (samuelsramko) [inline src/model/request/egld/EsdtToken.ts:6]: make Enum of this

File: src/transaction/egld.ts [hunk 3]
```diff
@@ -0,0 +1,992 @@
...
+
+    return JSON.stringify(tx);
+};
+
+/**
+ * Returns EGLD server to connect to.
+ * @param provider url of the EGLD Server to connect to. If not set, default public server will be used.
+ * @param fromPrivateKey optional private key of sender account
+ */
+export const getEgldClient = (provider?: string, fromPrivateKey?: string) => {
```
Comment (samuelsramko) [inline src/transaction/egld.ts:102]: if you just return string, private key doesn't have to be here.

**Review by samuelsramko — CHANGES_REQUESTED** [review]

File: src/transaction/egld.ts (hunk 3 above)
Comment (isra67) [reply src/transaction/egld.ts:102]: ok

File: src/model/request/egld/EgldEsdtTransaction.ts (hunk 1 above)
Comment (isra67) [reply src/model/request/egld/EgldEsdtTransaction.ts:28]: ok

File: src/model/request/egld/EsdtToken.ts (hunk 2 above)
Comment (isra67) [reply src/model/request/egld/EsdtToken.ts:6]: ok

**Review by samuelsramko — APPROVED** [review]

**Pull request merged**